horovod/spark/keras/datamodule.py

```diff
@@ -83,9 +83,9 @@ def __enter__(self):
 
     def __exit__(self, type, value, traceback):
         if self.has_val and self.val_reader:
-            self.val_reader.stop()
+            self.val_reader.__exit__(type, value, traceback)
```

Comment:
why `stop()` is removed her?

Reply:
This is trying to leverage the `val_reader` exit handler, which in some cases do more than just `stop()`, e.g. [Petastorm Reader](https://github.com/uber/petastorm/blob/master/petastorm/reader.py#L707-L709) also invokes `self.join()`, which can cause some race conditions on exit if not called.